test/src/main/java/org/axonframework/test/matchers/Matchers.java

```diff
@@ -155,6 +155,17 @@ public static Matcher<List<CommandMessage<?>>> noCommands() {
         return new EmptyCollectionMatcher<>("commands");
     }
 
+    /**
+     * Returns a Matcher that matches with exact class type defined by the given {@code expected}.
+     *
+     * @param expected The expected event class
+     * @param <T>      The type of event to match against
+     * @return a matcher that matches based on the class
+     */
+    public static <T> ExactClassMatcher<T> exactClassOf(Class<T> expected) {
```

Comment:
I'd change the return type of this method to `Matcher`. By doing so, we have some freedom to switch the actual returned matcher, if the future requires us to.

Reply:
Agree